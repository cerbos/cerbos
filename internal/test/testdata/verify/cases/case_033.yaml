description: Default policy version set on the suite
