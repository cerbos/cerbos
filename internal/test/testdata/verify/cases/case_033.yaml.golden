{
  "suites": [
    {
      "file": "suite_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_FAILED",
        "testsCount": 3,
        "resultCounts": [
          {
            "result": "RESULT_PASSED",
            "count": 1
          },
          {
            "result": "RESULT_FAILED",
            "count": 2
          }
        ]
      },
      "testCases": [
        {
          "name": "John and his leave request",
          "principals": [
            {
              "name": "john",
              "resources": [
                {
                  "name": "john_leave_request",
                  "actions": [
                    {
                      "name": "view:public",
                      "details": {
                        "result": "RESULT_FAILED",
                        "failure": {
                          "expected": "EFFECT_ALLOW",
                          "actual": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "approve",
                      "details": {
                        "result": "RESULT_PASSED",
                        "success": {
                          "effect": "EFFECT_DENY"
                        }
                      }
                    },
                    {
                      "name": "defer",
                      "details": {
                        "result": "RESULT_FAILED",
                        "failure": {
                          "expected": "EFFECT_ALLOW",
                          "actual": "EFFECT_DENY"
                        }
                      }
                    }
                  ]
                }
              ]
            }
          ]
        }
      ],
      "description": "Tests for verifying something"
    }
  ],
  "summary": {
    "overallResult": "RESULT_FAILED",
    "testsCount": 3,
    "resultCounts": [
      {
        "result": "RESULT_PASSED",
        "count": 1
      },
      {
        "result": "RESULT_FAILED",
        "count": 2
      }
    ]
  }
}
